src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Repositories/PolicyInformationRepository.cs

```diff
@@ -0,0 +1,234 @@
+                Application application = await _client
+                .ReadDocumentAsync<Application>(
+                    uri,
+                    new RequestOptions { PartitionKey = new PartitionKey(org) });
+                PostProcess(application);
+
+                return application;
+            }
+            catch (DocumentClientException ex)
+            {
+                return null;
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex.Message);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// postprocess applications so that appId becomes org/app-23 to use outside cosmos
+        /// </summary>
+        /// <param name="application">the application to postprocess</param>
+        private void PostProcess(Application application)
+        {
+            application.Id = CosmosIdToAppId(application.Id);
+        }
+
+        /// <summary>
+        /// Converts the appId "{org}/{app}" to "{org}-{app}"
+        /// </summary>
+        /// <param name="app">Application identifier which is unique within an organisation.</param>
+        /// <returns>the converted id</returns>
+        private string AppIdToCosmosId(string app)
+        {
+            string cosmosId = app;
+
+            if (app != null && app.Contains("/"))
+            {
+                string[] parts = app.Split("/");
+
+                cosmosId = $"{parts[0]}-{parts[1]}";
+            }
+
+            return cosmosId;
+        }
+
+        /// <summary>
+        /// Converts the cosmosId "{org}-{app}" to "{org}/{app}"
+        /// </summary>
+        /// <param name="cosmosId">the id to convert</param>
+        /// <returns>the converted id</returns>
+        private string CosmosIdToAppId(string cosmosId)
+        {
+            string appId = cosmosId;
+
+            int firstDash = cosmosId.IndexOf("-");
+
+            if (firstDash > 0)
+            {
+                string app = cosmosId.Substring(firstDash + 1);
+                string org = cosmosId.Split("-")[0];
+
+                appId = $"{org}/{app}";
+            }
+
+            return appId;
+        }
+
+        /// <summary>
+        /// Converts the instanceId (id) of the instance from {instanceGuid} to {instanceOwnerId}/{instanceGuid} to be used outside cosmos.
+        /// </summary>
+        /// <param name="instance">the instance to preprocess</param>
+        private void PostProcess(Instance instance)
+        {
+            instance.Id = $"{instance.InstanceOwnerId}/{instance.Id}";
+        }
+
+        /// <summary>
+        /// An instanceId should follow this format {int}/{guid}.
+        /// Cosmos does not allow / in id.
+        /// But in some old cases instanceId is just {guid}.
+        /// </summary>
+        /// <param name="instanceId">the id to convert to cosmos</param>
+        /// <returns>the guid of the instance</returns>
+        private string InstanceIdToCosmosId(string instanceId)
```

Comment:
Could we consider renaming to instanceIdToInstanceGuid? Makes it a bit more intuitive. My understanding would be that cosmosId i {instanceOwner}-{instanceGuid} as this is the terminology we use for an app.